lib/dams/controller_helper.rb

```diff
@@ -898,28 +898,45 @@ def referrer_controller( request )
       end
     end
 
-  def metadata_display?(data)
-    result = false
-    return result unless data
-    data.each do |datum|
-      result = true if datum.include?('localDisplay') || datum.include?('metadataDisplay')
+    def metadata_display?(data)
```

Comment:
Since we already have this `metadata_display?` method, and it appears to be used in the catalog_helper and dams_resource_controller, do we need the `local_display?` and `metadata_only_display?` methods? Could we use this instead? This one seems to just check both (though it could be rewritten to use `any?` also like the others. Just a thought.

Reply:
@mcritchlow I think we may need to a separate method for metadata_only_display? to determine the different behavior with localDisplay and metadataDisplay objects. For localDisplay objects,  they allow downloading of files (video/audio excluded) while metadataDisplay won't allow file download for any files.

Reply:
@lsitu - Ok, thanks for clarifying. Perhaps I misunderstood what Greg said. I thought he said if either were filled in then one could not download derivatives. But yeah if there are distinctions then I guess that's another issue. Poorly named properties it turns out, because they don't actually describe what they do :)

Reply:
@mcritchlow Yes, it's confusing to me too. So I asked Greg to clarify the localDisplay thing earlier since the access control issue is always sensitive to us. Do you want to rename it?

Reply:
@lsitu - I'm not sure. Honestly, all your tests pass and I think it's a pretty comprehensive test suite. I don't want to slow this down more than necessary. I just think the data model implications and application to these access control scenarios is unsustainable, very fragile at the least. But that's hopefully an issue we can address in the next system. Perhaps unless @hweng or @VivianChu have any other concerns, I'll just merge this and we can move on? (Thanks for all your awesome work on this complicated ticket)

Reply:
@mcritchlow Could you hold on? I'll refactor it a little bit. Thanks.